core/factorization/ic.cpp

```diff
@@ -79,15 +103,75 @@ std::unique_ptr<Composition<ValueType>> Ic<ValueType, IndexType>::generate(
     exec->run(ic_factorization::make_add_diagonal_elements(
         local_system_matrix.get(), false));
 
+    std::shared_ptr<const matrix_type> ic;
     // Compute LC factorization
-    exec->run(ic_factorization::make_compute(local_system_matrix.get()));
+    if (std::dynamic_pointer_cast<const OmpExecutor>(exec) ||
+        parameters_.algorithm == factorize_algorithm::syncfree) {
+        std::unique_ptr<gko::factorization::elimination_forest<IndexType>>
```

Comment:
I think it would make sense we use the syncfree as default not sparselib.
some default value not working sounds a bit weird to me.

Reply:
I think ginkgo can be kind of sparselib for itself? We have that for reference check.

Reply:
Personally, I'm not too keen on having "sparselib" mean two different things: an external library for CUDA/HIP, and Ginkgo itself for OpenMP(/Reference). Also I realize I got my comment wrong above: I meant "if you tried to NOT use `syncfree`". 
I guess my main concern is the fact that the parameter is basically meaningless for OpenMP -- but if you use it, you don't get a warning that you selected a meaningless option. Right now, if you try to create an Ilu/Ic preconditioner with OpenMP, it will tell you it's not implemented, right? Why not let it continue to do that in the default case (`sparselib`)?

I agree it would make more sense to have Ginkgo's version as the default once it's added, though I guess it's a bit tricky to change default behavior on users...

Reply:
Yes, I have raised it in our meeting. We keep the original behavior: default is stillsparselib, but omp will failed.
We throw a descriptive error message to indicate user should use syncfree for omp executor.
I have updated it